drivers/power/ltc3350/ltc3350.c

```diff
@@ -0,0 +1,371 @@
+
+	if (ret) {
+		*read_data = 0;
+			
+	} else {
+		*read_data = (rx_buff[1] << 8) | rx_buff[0];
+	}
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Writes data to the device (I2C)
+ *
+ * @param dev          - The device structure.
+ * @param base_address - Address of the base register.
+ * @param size         - The number of bytes to be written. It is the size
+ *                       of the write_data value in bytes.
+ * @param write_data   - The data which is going to be written.
+ *
+ * @return ret         - Result of the writing procedure.
+*******************************************************************************/
+int ltc3350_write_device_data(struct ltc3350_dev *dev, uint8_t base_address,
+			      uint16_t write_data)
+{
+	int ret = 0;
+	uint8_t txbuff[3] = {0};
+
+	txbuff[0] = base_address;
+	txbuff[1] = write_data & 0xFF;
+	txbuff[2] = write_data >> 8;
+
+	ret = no_os_i2c_write(dev-> i2c_desc, txbuff, 3, 1);
+
+	return ret;
+}
+
+#if	LTC3350_USE_MEASUREMENTS
+
+/***************************************************************************//**
+ * @brief  Gets number of capacitors configured based on PIN
+ *
+ * @param dev          - The device structure.
+ * @param num_caps	   - Number of capacitors configured.
+ *                       00-1 cap, 01 - 2 caps, 10 - 3caps and 11 - 4 caps
+ *
+ * @return ret         - Result of the reading procedure.
+*******************************************************************************/
+int ltc3350_get_num_caps(struct ltc3350_dev *dev, uint8_t *num_caps)
+{
+	int ret = 0;
+	uint16_t value;
+
+	ret = ltc3350_read_device_data(dev, LTC3350_AD_NUM_CAPS, &value);
+	if (ret)
+		return ret;
+
+	*num_caps = value +1; //check num_caps param
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Reads the ESR data.
+ *
+ * @param dev          - The device structure.
+ * @param value 	   - Measured ESR.
+ *
+ * @return ret         - Result of the reading procedure.
+*******************************************************************************/
+int ltc3350_get_esr(struct ltc3350_dev *dev, uint16_t *value)
+{
+	int ret = 0;
+	uint16_t read_value;
+
+	ret = ltc3350_read_device_data(dev, LTC3350_AD_MEAS_ESR, &read_value);
+	if (ret)
+		return	ret;
+
+	*value = read_value;
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Reads the Vcap data for selected capacitor
+
+ * @param dev          - The device structure.
+ * @param n_cap 	   - Number of capacitor to read, 0 for stack voltage
+ * @param value		   - The read value
+ *
+ * @return ret         - Result of the reading procedure.
+*******************************************************************************/
+int ltc3350_get_vcap(struct ltc3350_dev *dev, uint8_t n_cap, uint16_t *value)
+{
+	int ret = 0;
+
+	switch(n_cap) {
+
+	case 0:
+
```

Comment:
nitpick: drop these empty lines in the switch..case statement. there are not needed. also drop those after the break; statement.

Reply:
You could also use if statements, if you want the code to be less verbose. Something like:

if (n_cap > 4)
	return -EINVAL;

if (!n_cap)
	return ltc3350_read_device_data(dev, LTC3350_AD_MEAS_VCAP, value);
else
	return ltc3350_read_device_data(dev, LTC3350_AD_MEAS_VCAP1 + n_cap - 1, value);